keydata.go

```diff
@@ -590,60 +532,25 @@ func (d *keyData) validate(tpm *tpm2.TPMContext, policyUpdateData *keyPolicyUpda
 	// At this point, we know that the sealed object is an object with an authorization policy created by this package and with
 	// matching static metadata and persistent TPM resources.
 
-	if policyUpdateData == nil {
-		// If we weren't passed a private data structure, we're done.
-		return pcrPolicyCounterPub, nil
-	}
-
-	if policyUpdateData.version != d.version {
-		return nil, keyFileError{errors.New("mismatched metadata versions")}
-	}
-
-	// Verify that the private data structure is bound to the key data structure.
-	h := keyPublic.NameAlg.NewHash()
-	if _, err := tpm2.MarshalToWriter(h, policyUpdateData.creationData); err != nil {
-		panic(fmt.Sprintf("cannot marshal creation data: %v", err))
-	}
-
-	if _, _, err := tpm.CertifyCreation(nil, keyContext, nil, h.Sum(nil), nil, policyUpdateData.creationTicket, nil,
```

Comment:
are we losing integrity/security without this when we still use v0 update files?

Reply:
This was only there for integrity checking of the file containing the authorization key, in case it grew to contain additional fields. It's not required for checking the actual authorization key - there's already a check to ensure that the public part of it matches the authorization policy of the sealed key, and go's crypto library performs validation that the public and private parts are consistent when it is loaded. Since the file is being removed, there's not much point keeping this additional complexity around.

Reply:
thanks for the clarification on this